src/hotspot/share/runtime/threadHeapSampler.cpp

```diff
@@ -396,10 +396,11 @@ void ThreadHeapSampler::pick_next_geometric_sample() {
       (0.0 < log_val ? 0.0 : log_val) * (-log(2.0) * (get_sampling_interval())) + 1;
   assert(result > 0 && result < static_cast<double>(SIZE_MAX), "Result is not in an acceptable range.");
   size_t interval = static_cast<size_t>(result);
-  _bytes_until_sample = interval;
+  _sample_threshold = interval;
```

Comment:
Nit: The line 399 is not needed as the value is reset at 400.

Reply:
Nice catch. This is debugging code from the early stage of this patch. It set up a consistent state to make it easier to get consistent data from the test runs. This also means that we need to respin our testing to make sure that the randomization doesn't trigger anything that we didn't catch with the fixed _sample_threshold.

Reply:
The unit tests still pass after the fix for this.